CHANGELOG.md

```diff
@@ -4,6 +4,36 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.23.0] - XXXX-XX-XX
+
+### Fixed
+* Array alignment problem for stack arrays allocated for kernel arguments. (#1357)
+* Issue #892, #906 caused by incorrect code generation for indexing (#1377)
+* Fix `KernelHasReturnValueError` inside `KernelDispatcher`. (#1394)
+* Issue #1390: broken support for slicing into `dpctl.tensor.usm_ndarray` in kernels (#1425)
+
+### Added
+* A new overloaded `dimensions` attribute for all index-space id classes (#1359)
+* Support for `AtomicRef` creation using multi-dimension arrays (#1367)
+* Support for linearized indexing functions inside a JIT compiled kernel (#1368)
+* Improved documentation: overview (#1341), kernel programming guide (#1388), API docs (#1414), configs options (#1415), comparison with SYCL API (#1417)
+* New `PrivateArray` class in `kernel_api` to replace `dpex.private.array` (#1370, #1377)
+* Support for libsycinterface::DPCTLKernelArgType enumfor specifying type of kernel args instead of hard coding (#1382)
+* New indexing unit tests for kernel_api simulator and JIT compiled modes (#1378)
+* New unit tests to verify all `kernel_api` features usable inside `device_func` (#1391)
+* A `sycl::local_accessor`-like API (`kernel_api.LocalAccessor`) for numba-dpex kernel (#1331)
```

Comment:
Perhaps this should warn that this feature requires `dpctl` >=0.16.1

Reply:
Good idea